core/src/main/java/kafka/server/ReplicaFetcherTierStateMachine.java

```diff
@@ -68,14 +68,16 @@
  */
 public class ReplicaFetcherTierStateMachine implements TierStateMachine {
     private static final Logger log = LoggerFactory.getLogger(ReplicaFetcherTierStateMachine.class);
-
+    private  BrokerTopicStats brokerTopicStats;
     private LeaderEndPoint leader;
     private ReplicaManager replicaMgr;
 
     public ReplicaFetcherTierStateMachine(LeaderEndPoint leader,
-                                          ReplicaManager replicaMgr) {
+                                          ReplicaManager replicaMgr,
+                                          BrokerTopicStats brokerTopicStats) {
```

Comment:
you can get  brokerTopicStats as replicaMgr.brokerTopicStats()

Reply:
Thank you for pointing this out! So I don't need to change as many classes. Fixed now

Reply:
Note that this would currently break a lot of tests because `MockTierStateMachine` passes `null` for the `replicaMgr` parameter.